src/meltano/core/setting_definition.py

```diff
@@ -211,6 +225,7 @@ def __init__(
 
         kind = SettingKind(kind) if kind else None
         hidden = hidden or kind is SettingKind.HIDDEN or None
+        sensitive = sensitive or kind in SettingKind.get_sensitive() or None
```

Comment:
`SettingKind.get_sensitive()` is called for every setting. Should it be cached?

Reply:
How do you go about caching a `classmethod`? I only see uses of `@cache_property`. It got me wondering if there was a better way to do this though - something like:

```py
class SettingKind(YAMLEnum):
    """Supported setting kinds."""

    STRING = "string"
    INTEGER = "integer"
    BOOLEAN = "boolean"
    DATE_ISO8601 = "date_iso8601"
    EMAIL = "email"
    PASSWORD = "password"  # noqa: S105
    OAUTH = "oauth"
    OPTIONS = "options"
    FILE = "file"
    ARRAY = "array"
    OBJECT = "object"
    HIDDEN = "hidden"

    @property
    def is_sensitive(self):
        return self in {
            SettingKind.PASSWORD,
            SettingKind.OAUTH,
        }
```

```
        sensitive = sensitive or kind and kind.is_sensitive
```

I don't know if that just pushes the problem somewhere else though?

Reply:
@ReubenFrankel Yeah, I like the suggestion of having `SettingKind.is_sensitive` and it can be probably be cached with `functools.cached_property`.

Reply:
https://github.com/meltano/meltano/pull/7874/commits/bed9e9e2207fe1acb14d0e8da996f2bd99adfc56